subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/JavaRecompilationSpecProvider.java

```diff
@@ -132,26 +131,12 @@ private void processOtherChanges(CurrentCompilation current, PreviousCompilation
                 if (emptyAnnotationProcessorPath) {
                     continue;
                 }
-                String changeName = determineChangeName(fileChange.getChangeType());
-                spec.setFullRebuildCause(fileChange.getFile().getName() + " has been " + changeName, null);
+                spec.setFullRebuildCause(rebuildClauseForChangedNonSourceFile("resource", fileChange), null);
                 return;
             }
         }
     }
 
-    private String determineChangeName(ChangeType changeType) {
-        switch (changeType) {
-            case ADDED:
-                return "added";
-            case REMOVED:
-                return "removed";
-            case MODIFIED:
-                return "modified";
-            default:
-                throw new AssertionError("Unknown change type: " + changeType);
-        }
-    }
-
     private static String getClassNameForRelativePath(String relativePath) {
         return relativePath.replace('/', '.').replaceAll("\\.java$", "");
     }
```

Comment:
This looks like we should have it in some better shared location. (We also have something similar in `JvmSourceFile`.)

Reply:
`JvmSourceFile` is only a test fixture. Where else do we share this code?

Reply:
Seemed like a piece of code that would be useful in general. We can leave it like this, too.